crates/router/src/connector/stax.rs

```diff
@@ -88,28 +81,277 @@ impl ConnectorCommon for Stax {
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::Total(total_error) => total_error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::Error(error) => error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::Authorization(authorization_error) => authorization_error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::Firstname(error) => error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::Lastname(error) => error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::Email(error) => error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::Company(error) => error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::CardNumber(error) => error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::PersonName(error) => error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::CustomerId(error) => error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::Meta(error) => error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::PaymentMethodId(error) => error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+            stax::StaxErrorResponseTypes::TokenInvalid(token_invalid_error) => token_invalid_error
+                .first()
+                .unwrap_or(&consts::NO_ERROR_MESSAGE.to_string())
+                .to_owned(),
+        };
 
         Ok(ErrorResponse {
             status_code: res.status_code,
-            code: response.code,
-            message: response.message,
-            reason: response.reason,
+            code: consts::NO_ERROR_CODE.to_string(),
```

Comment:
Can we have "StaxErrorResponseTypes" as error code?

Reply:
Done